src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -904,7 +904,6 @@ private void fetchFrom(GitClient git,
                 for (GitSCMExtension extension : extensions) {
                     extension.decorateFetchCommand(this, run, git, listener, fetch);
                 }
-                fetch.execute();
```

Comment:
Upon the removal of fetch.execute(), if `git.hasGitRepo()` is true which means that we are "Fetching changes from the remote Git repository", how would GitSCM _retrieve_ the changes in the remote repo?

Reply:
I would have checked other cases also. I will resubmit some changes.